packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -758,17 +784,41 @@ export default class metadataBackup {
   async restoreMetadataBackup(id: string) {
     const app = this._app
     const logger = this._logger
-    const [remoteId, dir, ...path] = id.split('/')
-    const metadataFolder = `${dir}/${path.join('/')}`
+    const [remoteId, ...path] = id.split('/')
+    const backupId = path.join('/')
 
-    const { proxy, url, options } = await app.getRemoteWithCredentials(remoteId)
+    const remote = await app.getRemoteWithCredentials(remoteId)
+    const { type, poolUuid } = parseMetadataBackupId(backupId)
 
     let rootTaskId
+    const localTaskIds = { __proto__: null }
+    const onLog = async log => {
+      if (type === 'xoConfig' && log.parentId === undefined && log.status === 'success') {
```

Comment:
`end` task does not have `parentId`, you need to check `log.taskId === rootTaskId`:
```suggestion
      if (type === 'xoConfig' && log.taskId === rootTaskId && log.status === 'success') {
```